enterprise/server/remote_execution/containers/podman/podman.go

```diff
@@ -113,19 +117,29 @@ type pullStatus struct {
 }
 
 type Provider struct {
-	env                   environment.Env
-	imageCacheAuth        *container.ImageCacheAuthenticator
-	buildRoot             string
-	imageStreamingEnabled bool
-	regClient             regpb.RegistryClient
-	optImageCache         *optImageCache
+	env                     environment.Env
+	imageCacheAuth          *container.ImageCacheAuthenticator
+	buildRoot               string
+	imageStreamingSupported bool
+	regClient               regpb.RegistryClient
+	optImageCache           *optImageCache
 }
 
 func NewProvider(env environment.Env, imageCacheAuthenticator *container.ImageCacheAuthenticator, buildRoot string) (*Provider, error) {
 	c, err := newOptImageCache()
 	if err != nil {
 		return nil, err
 	}
+
+	if *imageStreamingEnabled {
+		if *imageStreamingRegistryGRPCTarget == "" {
+			return nil, status.FailedPreconditionError("registry gRPC target must be set to use image streaming")
+		}
+		if *imageStreamingRegistryHTTPTarget == "" {
+			return nil, status.FailedPreconditionErrorf("registry HTTP target must be set to use image streaming")
+		}
+	}
+
```

Comment:
could you do something like setting a flag once you've checked things so the check only happens once?
```
    imageStreamingSupported := false
	if *imageStreamingEnabled {
		if *imageStreamingRegistryGRPCTarget == "" {
			return nil, status.FailedPreconditionError("registry gRPC target must be set to use image streaming")
		}
		if *imageStreamingRegistryHTTPTarget == "" {
			return nil, status.FailedPreconditionErrorf("registry HTTP target must be set to use image streaming")
		}
        imageStreamingSupported = true
	}
```


Then below you can remove the:

```
imageStreamingSupported := *imageStreamingRegistryHTTPTarget != "" && *imageStreamingRegistryGRPCTarget != ""
```

Reply:
Done.